### ISSUE TITLE
Investigate what anonymous ID in RS for kids

### ISSUE THREAD

**Issue by mulundapm** [bcc-code/bcc-media-app#418]

**Comment by andreasgangso** [comment]
**Summary**
AnonymousId used to be based on the deviceID. This was changed in august/september last year, in rudder_sdk_flutter version 2.5.0, which we upgraded to september 6th 2023 (https://github.com/bcc-code/bcc-media-app/commit/999a416691ddc66e37e5d49895b2c7405cd8ea90#diff-e125d55a387d4b24048735f9ca6508c4e860f5ca1f3242edfd9df27e7d8a7af0R1392)

It's hard to follow their source code for anonymousId on android especially, but as far as I can tell, the "Traits anonymousid" and "anonymousId" is supposed to be the exact same thing. I have no idea _why_ they want to send it duplicated, but I guess it doesnt matter.

**Comment by andreasgangso** [comment]
**History**
AnonymousId used to be == deviceID.
For kids, since rudderstack hadn't implemented a way to disable this, I implemented a workaround via a fork: https://github.com/andreasgangso/rudder-sdk-android/commit/ef8d0919d196d2e9e39cbdd354410294d88e26d0
As far as I understand, this made the anonymousId unique every launch. This was included ever since the initial release of the kids app.

Later on RS implemented a separation of deviceId and anonymousId, available in rudder_sdk_flutter version 2.5.0, which we upgraded to in Kids v4.9.0 (https://github.com/bcc-code/bcc-media-app/commit/999a416691ddc66e37e5d49895b2c7405cd8ea90#diff-e125d55a387d4b24048735f9ca6508c4e860f5ca1f3242edfd9df27e7d8a7af0R1392). From that point on, the anonymousId would always be a UUID and it would be saved for the lifetime of that install.

**Traits anonymousId vs anonymousId**
It's hard to follow their source code for anonymousId on android especially, but as far as I can tell, the "Traits anonymousid" and "anonymousId" is supposed to be the exact same thing. I have no idea _why_ they want to send it duplicated, but I guess it doesnt matter.

**Conclusion**
hm
